src/store/metadata/stub/validation.py

```diff
@@ -0,0 +1,423 @@
+
+            Defined topics from topics.json:
+            {topic_ids}"""
+            )
+
+
+def validate_publishers(
+    publishers: Dict, dataset_publishers: Set, dataset_creators: Set
+):
+    """
+    Confirm that:
+    - any publisher listed at the `datasets` level exists
+    - any creator listed at the `datasets` level exists
+    - all publisher resources appear in one or more of the above checks
+    """
+    _validate_resource_count(publishers, "publishers")
+    publisher_ids = [publisher["@id"] for publisher in publishers["publishers"]]
+    publishers_referenced = []
+
+    for publisher in dataset_publishers:
+        if publisher not in publisher_ids:
+            raise ValueError(
+                f"""
+            The publisher: {publisher}
+            is referenced in a datasets resource but does not exist in the definitions of publishers from publishers.json.
+            """
+            )
+        publishers_referenced.append(publisher)
+
+    for creator in dataset_creators:
+        if creator not in publisher_ids:
+            raise ValueError(
+                f"""
+            The creator {creator} is referenced in a datasets resource but does not exist in the definitions of publishers from publishers.json.
+            """
+            )
+        publishers_referenced.append(creator)
+
+    # At this point all publishers we've defined should have appeared at least once.
+    unique_publishers_referenced = list(set(publishers_referenced))
+
+    unique_publishers_referenced.sort()
+    publisher_ids.sort()
+
+    if len(unique_publishers_referenced) != len(publisher_ids):
+        raise ValueError(
+            f"""
+        Publishers are being created that do not appear to be utilised.
+
+        IDs of publisher resources being created:
+        {publisher_ids}
+
+        IDs of publisher resources being used:
+        {unique_publishers_referenced}
+        """
+        )
+
+    # Schema validation
+    schemas.Publishers(**publishers)
+
+
+def _assert_editions_ordered_by_issued(list_of_datasets, main_dict: str, sub_dict: str):
+    """
+    this function will assert that the list of datasets subtset
+    is ordered by issued, raises error if not
+    """
+
+    index = 0
+
+    list_of_issued = []
+
+    for dataset_dict in list_of_datasets[main_dict]:
+        if len(dataset_dict[sub_dict]) > 1 and dataset_dict[sub_dict] is not None:
+            for y in dataset_dict[sub_dict]:
+                if index == 0:
+                    first_value = dataset_dict[sub_dict][0]["issued"]
+                fornow = y["issued"]
+                list_of_issued.append(fornow)
+                index += 1
+
+            most_recent = max(list_of_issued)
+            assert (
+                first_value == most_recent
+            ), "The datasets should be ordered by 'issued' (from most recent)"
+
+
+def _assert_versions_ordered_by_issued(datasets, sub_dict: str):
```

Comment:
don't _think_ this is getting called either, think it needs to be called somewhere in `validate_editions()`

Reply:
Will delete - I rewrote it and added it to _validate_version_in_edition L175-L184